### PR TITLE
use consistent postcode formatting functions

### LINKED ISSUE #678: Consistent postcode formatting functions
There are a number of places in the code base where we are stripping whitespace out of postcodes, adding whitespace into postcodes, converting to uppercase, etc to deal with sources that use different formats. This leads to bugs e.g: PR #677

We need to rationalise this/deal with it more consistently.

### PR DESCRIPTION
**Pull request by chris48s**
Lets see if we can put an end to the postcode formatting madness

I've tested this but this change touches so many important things I bet it breaks something that I have not spotted. Hopefully if we do this now I will work out what that is when we're not in the grip of a really important election :)

Closes #678

### CODE REVIEW

File: polling_stations/apps/pollingstations/helpers.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import re
+
+
+def format_postcode_no_space(postcode):
+    return re.sub('[^A-Z0-9]', '', postcode.upper())
+
+
+def format_postcode_with_space(postcode):
+    pc = format_postcode_no_space(postcode)
+    return pc[:-3] + ' ' + pc[-3:]
```
Comment (symroe) [inline polling_stations/apps/pollingstations/helpers.py]: I've been meaning to ask this for a while: is the outcode _always_ 3 chars? Seems like an odd bit of consistency!

**Comment by symroe** [comment]
Might be worth adding some test cases what touch this? Or some dedicated cases for odd postcode inputs?

File: polling_stations/apps/pollingstations/helpers.py (hunk 1 above)
Comment (chris48s) [inline polling_stations/apps/pollingstations/helpers.py]: The bit before the space can be 2, 3 or 4 characters. As far as I'm aware, the last bit is always 3. Here's some links that agree with me:
https://www.getthedata.com/images/postcode_format.png
https://www.mrs.org.uk/pdf/postcodeformat.pdf

**Comment by chris48s** [comment]
Yeah the 2 new functions are implicitly covered by various existing tests (hence the coverage has increased even though I have not added tests) but I will update the PR with some explicit test cases for the 2 new functions tomorrow.

**Pull request merged**